apps/explorer/src/components/ownedobjects/views/FieldItem.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiMoveNormalizedType } from '@mysten/sui.js';
+
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import {
+    extractSerializationType,
+    getFieldTypeValue,
+    FieldValueType,
+} from '~/components/ownedobjects/utils';
+import { AddressLink, ObjectLink } from '~/ui/InternalLink';
+import { Link } from '~/ui/Link';
+import { Text } from '~/ui/Text';
+
+interface FieldItemProps<T> {
+    value: T;
+    type: SuiMoveNormalizedType;
+    truncate?: boolean;
+}
+
+export function FieldItem<T>({
+    value,
+    type,
+    truncate = false,
+}: FieldItemProps<T>) {
+    // for object types, use SyntaxHighlighter
+    if (typeof value === 'object') {
+        return (
+            <SyntaxHighlighter
+                code={JSON.stringify(value, null, 2)}
+                language="json"
+            />
+        );
+    }
+
+    const normalizedType = extractSerializationType(type);
+    const moduleName = getFieldTypeValue(normalizedType, FieldValueType.MODULE);
```

Comment:
You probably could even normalize `extractSerializationType` down to just a string so that you don't need two functions